src/core/proto_reader/reader.rs

```diff
@@ -108,6 +115,38 @@ impl ProtoReader {
         Ok(descriptors_vec)
     }
 
+    /// Used to resolve dependencies proto files using file reader
+    async fn file_resolve(
+        &self,
+        parent_proto: FileDescriptorProto,
+        parent_path: Option<&Path>,
+    ) -> anyhow::Result<Vec<FileDescriptorProto>> {
+        let parent_path = parent_path.map(|p| p.to_path_buf());
+
+        self.resolve_dependencies(parent_proto, move |import| {
+            let parent_path = parent_path.clone();
+            let self_ref =
+                ProtoReader { reader: self.reader.clone(), runtime: self.runtime.clone() };
+
+            async move { self_ref.read_proto(import, parent_path.as_deref()).await }.boxed()
+        })
+        .await
```

Comment:
```suggestion
        self.resolve_dependencies(parent_proto, |import| {
            let parent_path = parent_path.map(|p| p.to_path_buf());
            let this = self.clone();

            async move { this.read_proto(import, parent_path.as_deref()).await }.boxed()
        })
        .await
```

derive Clone on self instead of cloning manually.

Reply:
If I do this, `this` variable is `&ProtoReader` instead of `ProtoReader.` And I get a lifetime error when it is moded into the async block.

Reply:
you need a `#[derive(Clone)]` to do the cloning as mentioned, to use this

Reply:
Thanks